astropy/cosmology/units.py

```diff
@@ -50,8 +52,8 @@ def dimensionless_redshift():
     return u.Equivalency([(redshift, None)], "dimensionless_redshift")
 
 
-def redshift_temperature(cosmology=None, **atzkw):
-    """Convert quantities between redshift and CMB temperature.
+def redshift_distance(cosmology=None, distance_kind="comoving", **atzkw):
```

Comment:
No reason to delay, but maybe just "kind"?

Reply:
Yeah. That does sound better. And then it's less similar to ``distance=`` in ``with_redshift()``.

Reply:
Renamed.